### PR TITLE
Baseline Model depends upon #21

### PR DESCRIPTION
**Pull request by philipcalderwood**
Baseline model as discussed on call. Draft PR given changes to me made. Adapted from MIT licensed torch implementation of UNet. https://github.com/jvanvugt/pytorch-unet

### CODE REVIEW

File: baseline_model_loader.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+# Proper transformation for segmentation masks
+# Modified target transform for segmentation masks
+target_transform = transforms.Compose(
+    [
+        # Resize using nearest neighbor to preserve class values
+        transforms.Resize(
+            (256, 256), interpolation=transforms.InterpolationMode.NEAREST
+        ),
+        # Use the numpy path instead of ToTensor()
+        lambda x: torch.from_numpy(np.array(x)).long().unsqueeze(0),
```
Comment (philipcalderwood) [inline baseline_model_loader.py:30]: I ran into problems with losing the integer classification 1,2,3. This was one potential problem with directly casting to Tensor - we lose the integer classification necessary for segmentation. Happy to take guidance on doing this cleaner?

Comment (timfarkas) [reply baseline_model_loader.py:30]: hi not sure if this is still relevant,
i don't quite understand what 'losing the integer classification' means?
did you try creating the loaders using James' data.create_dataloaders?

**Comment by timfarkas** [comment]
looks good to me rn after a quick glance! 
would be great if you could add the model to models.py, and then just have one script 'baseline_segmentation.py' that handles training and evaluation!

**Comment by philipcalderwood** [comment]
Thanks for the review Tim. I wasnt able to get classification to work at
one point - I thought it was due to floats. I’ll try with the dataloader
tomorrow and update.

**Comment by philipcalderwood** [comment]
i'm really sorry the lines changed is so large - it is because of the format on save I was using. do we have a group formatter I could apply perhaps to realign with main?

File: utils.py [hunk 2]
```diff
@@ -5,6 +5,7 @@
 from torch.utils.data import TensorDataset, DataLoader
 import os
 
+
```
Comment (philipcalderwood) [inline utils.py:8]: formatting changes unfortunately

File: utils.py
Comment (philipcalderwood) [inline utils.py:1]: formatting changes aside from one new fn - segmentation accuracy

File: pre_training.py
Comment (philipcalderwood) [inline pre_training.py:1]: temporarily modified this to suit my model - I removed the list of models. I will fix this before final integration

File: data.py
Comment (philipcalderwood) [inline data.py:1]: uses code from James PR
